File: Frameworks/RSCore/RSCore/NSImage+RSCore.m
//
//  NSImage+RSCore.m
//  RSCore
//
//  Created by Brent Simmons on 3/25/15.
//  Copyright (c) 2015 Ranchero Software, LLC. All rights reserved.
//

#import "NSImage+RSCore.h"


@implementation NSImage (RSCore)


+ (void)rs_imageWithData:(NSData *)data imageResultBlock:(RSImageResultBlock)imageResultBlock {

	NSParameterAssert(data != nil);

	dispatch_async(dispatch_get_global_queue(QOS_CLASS_DEFAULT, 0), ^{
		NSImage *image = [[NSImage alloc] initWithData:data];
		RSCallBlockWithParameter(imageResultBlock, image);
	});
}


+ (instancetype)imageWithContentsOfFile:(NSString *)f {

	return [[self alloc] initWithContentsOfFile:f];
}


@end

File: Modules/AppKitExtras/Sources/AppKitExtras/NSImage+RSCore.swift
//
//  NSImage+RSCore.swift
//  RSCore
//
//  Created by Brent Simmons on 12/16/17.
//  Copyright © 2017 Ranchero Software, LLC. All rights reserved.
//
#if os(macOS)
import AppKit

public extension NSImage {

	func tinted(with color: NSColor) -> NSImage {

		let image = self.copy() as! NSImage

		image.lockFocus()

		color.set()
		let rect = NSRect(x: 0, y: 0, width: image.size.width, height: image.size.height)
		rect.fill(using: .sourceAtop)

		image.unlockFocus()

		image.isTemplate = false
		return image
	}
}
#endif
